icx-proxy/src/main.rs

```diff
@@ -187,10 +248,49 @@ async fn forward_request(
             }
         }
 
-        Ok(builder.body(body)?)
+        builder.body(body)?
     } else {
-        Ok(builder.body(http_response.body.into())?)
+        builder.body(http_response.body.into())?
+    };
+
+    if logger.is_trace_enabled() {
+        slog::trace!(
+            logger,
+            ">> {:?} {} {}",
+            &response.version(),
+            response.status().as_u16(),
+            response.status().to_string()
+        );
+
+        for (name, value) in response.headers() {
+            let value = String::from_utf8_lossy(value.as_bytes());
+            slog::trace!(logger, ">> {}: {}", name, value);
+        }
+
+        let body = body.unwrap_or_else(|| b"... streaming ...".to_vec());
+
+        slog::trace!(logger, ">>");
+        slog::trace!(
+            logger,
+            ">> {}{}",
+            match std::str::from_utf8(&body) {
+                Ok(s) => format!(
+                    r#""{}""#,
+                    s[..usize::min(MAX_LOG_BODY_SIZE, s.len())].escape_default()
+                ),
```

Comment:
Is `format` necessary here?
```suggestion
                Ok(s) => s[..usize::min(MAX_LOG_BODY_SIZE, s.len())].escape_default(),
```